src/QsCompiler/CompilationManager/FileContentManager.cs

```diff
@@ -32,6 +32,11 @@ public class FileContentManager : IDisposable
 
         public string FileName { get; }
 
+        /// <summary>
+        /// Indicates the current version of the document according to the Language Server Protocol.
+        /// </summary>
+        public int? Version { get; set; }
```

Comment:
I don't think `Version` can ever be `null`, can it?

Reply:
This field is storing a value that comes from `VersionedTextDocumentIdentifier.Version` which is a an `int?` ... I don't expect it to normally be `null`, but I wanted to preserve the type used by the Language Server Protocol. If we want to enforce it to not be null, we'll need to decide a behavior in case we ever receive a null. With this implementation, we're recording the version as reported by the LSP in all cases.

Reply:
Yeah, I saw that the LSP package we're using has it as `int?`, but the [LSP specification says that is it a non-nullable integer](https://microsoft.github.io/language-server-protocol/specifications/specification-current/#versionedTextDocumentIdentifier). If it is null, I think we are free to reject the message since it seems like it is invalid.

Reply:
We should use the same data type that the LSP package, since we don't know if they intend to update the implementation or the specification. But this is a very good point, and something I'll ask them in our next sync.

Reply:
As far as I know, the VS LSP package is not an official implementation of the LSP protocol - it is just the implementation that Visual Studio uses. I would trust the specification in case of any differences, especially since I recall that there have been other problems with nullable types in this package.